six/modules/c++/scene/source/ProjectionPolynomialFitter.cpp

```diff
@@ -102,6 +103,155 @@ ProjectionPolynomialFitter::ProjectionPolynomialFitter(
+        static_cast<double>(minRowI),
+        static_cast<double>(minColI));
+    types::RowCol<size_t> boundingExtent(maxRowI - minRowI + 1,
+                                         maxColI - minColI + 1);
+
+    // No offset for the polygon mask.
+    types::RowCol<sys::SSize_T> pmOffset(0,0);
+
+    // Get the PolygonMas. For each row of the polygon this will determine
+    // the first and last column of the row inside the convex hull of the
+    // polygon sent in.
+    polygon::PolygonMask polygonMask(polygon, fullExtent, pmOffset);
+    
+    // Compute a delta in the row direction as if the entire bounding row 
+    // extent will be covered by the point grid.
+    const double initialDeltaRow =
+        static_cast<double>(boundingExtent.row - 1) / (numPoints1D - 1);
+
+    // Scale factor for shrinking the row extent.
+    const double shrinkFactor = 0.1;
+
+    // Shring the row extent a bit.
+    const double deltaToRemove = initialDeltaRow * shrinkFactor;
+
+    // Get the new row start and end values.
+    const size_t newStartRow = static_cast<size_t>(
+        std::ceil(boundingOffset.row + deltaToRemove));
+    const size_t newEndRow = static_cast<size_t>(
+        std::floor(boundingOffset.row + boundingExtent.row - 1 -
+                   deltaToRemove));
+
+    // Check the new row extent.
+    if (newStartRow > newEndRow)
+    {
+        throw except::Exception(Ctxt(
+            "New bounding rectangle has no area"));
+    }
+
+    // Compute the row exent.
+    const size_t newExtentRow = (newEndRow - newStartRow + 1);
+    
+    // Compute the delta in the row direction for the new extent.
+    const double newDeltaRow =
+         static_cast<double>(newExtentRow - 1) / 
+         static_cast<double>(numPoints1D - 1);
+
+    double currentOffsetRow = static_cast<double>(newStartRow);
+    for (size_t ii = 0; ii < numPoints1D; ++ii, currentOffsetRow += newDeltaRow)
+    {
+        double currentRow = std::floor(currentOffsetRow);
+        size_t row = static_cast<size_t>(currentRow);
+
+        // Get the start column and number of columns inside the polygon row
+        // the current row.
+        const types::Range colRange = polygonMask.getRange(row);
+
+        // Check that there are internal points.
+        if (colRange.mNumElements == 0)
+        {
+            throw except::Exception(Ctxt(
+                "Column range has no elements"));
+        }
+
+        // Compute the delta in the column direction to cover the internal
+        // points.
+        const double newDeltaCol =
+            static_cast<double>(colRange.mNumElements - 1) /
+            static_cast<double>(numPoints1D - 1);
+
+        double currentCol = static_cast<double>(colRange.mStartElement);
+        for (size_t jj = 0; jj < numPoints1D; ++jj, currentCol += newDeltaCol)
+        {
+            const types::RowCol<double> currentOffset(currentRow, currentCol);
+
+            // Get the coordinate relative to the outPixelStart.
```

Comment:
Lines 239-250 seem like they should be able to be put into a function that both constructors use instead of having two copies